monitoring_flows_and_extensions/index.md

```diff
@@ -0,0 +1,335 @@
+---
+title: 'Monitoring Flows & Extensions'
+description: 'We build a logging and monitoring pipeline for Directus applications without using any third-party dependencies.'
+author:
+  name: 'Daniel Band'
+  avatar_file_name: 'daniel.jpg'
+---
+Directus offers a flexible and extensible foundation for creating a wide range of applications. However, to address more complex use cases such as Backend as a Service or internal applications, we will develope a strategy for monitoring the application's state in this article.
+
+What benefits does monitoring give us:
+- Insights into the flow of data
+- Makes a production application debuggable
+- Allows us to listen to events and send out notifications via different channels
+
+## How is event monitoring different from activity logs and revision?
+ In the context of Directus, activity logs allow us to identify changes on a per-collection item basis, providing a trail to retrace the execution of workflows and view the input/output of operations. On the other hand, event logs focus on tracking specific pieces of data. The difference in granularity between activity tracking and event logging makes them complement each other. For instance, in a monitoring pipeline, when a specific event triggers an alert, we can use revisions to trace back the execution flow to analyze the context.
+
+## What events do we want to log?
+There are two main categories of events that serve different purposes.
+
+First, events that help us debug errors. These should contain information what, when and where something went wrong, accompanied by metadata to make the job of debugging easier. 
+
+The second type of events provides transactional data about the application state. Consider a scenario where we periodically import participants from an event ticketing app. In such cases, we want to track whether the import process started, the number of participants successfully imported, any import failures, and the reasons behind those failures. Event logs are instrumental in answering these questions without having to look at flow revisions.
+
+Furthermore, instead of sending email notifications directly from the location where an event occurs, we establish a separate workflow that listens for specific event types. This decoupling of alerting from our core business logic provides several advantages. It allows us to easily switch notification channels or transition to an external monitoring solution when the need arises, offering greater flexibility and adaptability in our alerting system.
```

Comment:
```suggestion
Instead of sending email notifications directly from the location where an event occurs, we can also establish a separate workflow that listens for specific event types. Decoupling alerting from our core business logic provides allows us to easily switch notification channels or transition to an external monitoring solution when the need arises, offering greater flexibility and adaptability in our alerting system.
```

Just making this a bit more succinct. I always ask myself in a review 'does this word add value for the time it takes to read and understand', as context to this suggestion.